packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1793,6 +1794,53 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     selectable.removeListener(_handleSelectableGeometryChange);
   }
 
+  @override
+  int? getContentLength() {
+    int length = 0;
+    for (final Selectable selectable in selectables) {
+      length += selectable.getContentLength() ?? 0;
+    }
+    return length;
+  }
+
+  @override
+  TextSelection? getLocalTextSelection() {
+    // initialize data structures
+    int start = 0;
+    int? end;
+    int numSelected = 0;
+    bool enteredSelectedRegion = false;
+
+    for (final Selectable selectable in selectables) {
```

Comment:
This should only loop through currentSelectionStartIndex to currentSelectionEndIndex

although you need to consider case where the selection can be inverse

Reply:
Since our goal is to determine a global offset, which is crucial in widgets that manage multiple selectables like the MultiSelectableSelectionContainerDelegate, it's important to account for all offsets, not just those that are selected.